packages/dx-chart-core/src/plugins/series/computeds.js

```diff
@@ -9,137 +9,146 @@ import {
-        .x(getX)
-        .y(getY);
-  }
-};
+export const dArea = area()
+  .x(getX)
+  .y1(getY)
+  .y0(getY1);
+
+export const dLine = line()
+  .x(getX)
+  .y(getY);
+
+export const dSpline = line()
+  .x(getX)
+  .y(getY)
+  .curve(curveCatmullRom);
 
 export const xyScales = (
-  domainsOptions,
-  argumentAxisName,
-  domainName,
-  layout,
-  stacks = [],
-  { groupWidth = 1, barWidth = 1 },
-) => {
-  const { width, height } = layout;
-  const argumentDomainOptions = domainsOptions[argumentAxisName];
-  const xScale = createScale(argumentDomainOptions, width, height, 1 - groupWidth);
-  const bandwidth = xScale.bandwidth
-    ? xScale.bandwidth()
-    : width / xScale.ticks().length;
-
-  return {
-    xScale,
-    yScale: createScale(domainsOptions[domainName], width, height),
-    x0Scale: createScale({
-      orientation: argumentDomainOptions.orientation,
-      type: 'band',
-      domain: stacks,
-    }, bandwidth, bandwidth, 1 - barWidth),
-  };
-};
+  argumentDomainOptions,
+  valueDomainOptions,
+  { width, height },
+  groupWidth,
+) => ({
+  xScale: createScale(argumentDomainOptions, width, height, 1 - groupWidth),
+  yScale: createScale(valueDomainOptions, width, height),
+});
 
 export const pieAttributes = (
-  valueField,
   data,
-  width,
-  height,
-  innerRadius,
-  outerRadius,
+  { xScale, yScale },
+  argumentField,
+  valueField,
 ) => {
+  const width = Math.max.apply(null, xScale.range());
+  const height = Math.max.apply(null, yScale.range());
   const radius = Math.min(width, height) / 2;
   const pieData = pie().value(d => d[valueField])(data);
 
-  return pieData.map(d => ({
-    d: arc().innerRadius(radius * innerRadius)
-      .outerRadius(radius * outerRadius || radius)
-      .startAngle(d.startAngle)
-      .endAngle(d.endAngle)(),
-    value: d.value,
+  return pieData.map(({
+    startAngle, endAngle, value, data: itemData,
+  }) => ({
+    d: arc()
+      .innerRadius(0)
+      .outerRadius(radius)
+      .startAngle(startAngle)
+      .endAngle(endAngle)(),
+    value,
+    data: itemData,
+    id: itemData[argumentField],
+    x: width / 2,
+    y: height / 2,
   }));
 };
 
 export const coordinates = (
   data,
-  scales,
+  { xScale, yScale },
   argumentField,
   valueField,
   name,
-) => computeLinePath(data, scales, argumentField, valueField, name);
+) => data.reduce((result, dataItem) => {
+  if (dataItem[argumentField] !== undefined && dataItem[valueField] !== undefined) {
+    return [...result, {
+      x: xScale(dataItem[argumentField]),
+      y: yScale(dataItem[`${valueField}-${name}-stack`][1]),
+      y1: yScale(dataItem[`${valueField}-${name}-stack`][0]),
+      width: xScale.bandwidth ? xScale.bandwidth() : 0,
+      id: dataItem[argumentField],
+      value: dataItem[valueField],
+    }];
+  }
+  return result;
+}, []);
 
-export const findSeriesByName = (name, series) => series.find(
-  seriesItem => seriesItem.uniqueName === name,
-);
+export const barCoordinates = (
```

Comment:
There are no tests on this method. It is the same with the dArea, dLine, dBar and dSpline methods.